python/fio/fio_plugin.py

```diff
@@ -0,0 +1,60 @@
+import sys
+import typing
+import json
+import subprocess
+from traceback import format_exc
+from typing import Union
+from pathlib import Path
+
+from arcaflow_plugin_sdk import plugin
+from fio_schema import (
+    FioJob,
+    FioSuccessOutput,
+    FioErrorOutput,
+    fio_output_schema,
+)
+
+
+@plugin.step(
+    id="workload",
+    name="fio workload",
+    description="run an fio workload",
+    outputs={"success": FioSuccessOutput, "error": FioErrorOutput},
+)
+def run(
+    params: FioJob,
+) -> typing.Tuple[str, Union[FioSuccessOutput, FioErrorOutput]]:
+    outfile_temp_path = Path("fio-plus.json")
+    infile_temp_path = Path("fio-input-tmp.fio")
+
+    try:
+        params.write_params_to_file(infile_temp_path)
+        cmd = [
+            "fio",
+            f"{infile_temp_path}",
+            "--output-format=json+",
+            f"--output={outfile_temp_path}",
+        ]
+
+        try:
+            subprocess.check_output(cmd)
+        except subprocess.CalledProcessError:
+            error_output: FioErrorOutput = FioErrorOutput(format_exc())
+            return "error", error_output
+
+        output: FioSuccessOutput = fio_output_schema.unserialize(
+            json.loads(outfile_temp_path.read_text())
+        )
+        if params.cleanup:
+            infile_temp_path.unlink(missing_ok=True)
+            outfile_temp_path.unlink(missing_ok=True)
+            Path(params.name + ".0.0").unlink(missing_ok=True)
```

Comment:
This is another place where a remnant file is getting left behind from the run.

Reply:
which file? In raw execution or the container?